src/main/java/org/cloudfoundry/promregator/lite/discovery/DiscoveryService.kt

```diff
@@ -0,0 +1,48 @@
+package org.cloudfoundry.promregator.lite.discovery
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.runBlocking
+import org.cloudfoundry.promregator.lite.config.PromregatorConfigurationV2
+import org.cloudfoundry.promregator.lite.discovery.SimpleAppInstanceScanner.Instance
+import org.springframework.stereotype.Service
+import java.util.concurrent.TimeUnit
+
+private val log = mu.KotlinLogging.logger {}
+
+@Service
+class DiscoveryService(
+        private val promregatorConfiguration: PromregatorConfigurationV2,
+        private val targetResolver: SimpleTargetResolver,
+        private val appInstanceScanner: SimpleAppInstanceScanner,
+) {
+    val cache = Caffeine.newBuilder()
```

Comment:
The new endpoint only uses a single cache. With the simplified CF calls, we won't need multiple cache layers.